Improve AddFirstProduct email note contents

ilyasfoo opened the pull request:
Fixes https://github.com/woocommerce/woocommerce-admin/issues/6363

This PR improves the contents of AddFirstProduct email note:

- Add merchant's name to greet (with default as "Hi there")
- Update heading
- Update image

It also changes `NotificationEmail` to utilize WC_Email's capability for placeholder contents in email notes by using the `format_string` class function. This allows for dynamic customization and personalization in email notes.

### Screenshots
![image](https://user-images.githubusercontent.com/3747241/111576606-c65d4e00-87eb-11eb-818c-2679e053ef20.png)

### Detailed test instructions:
- Checkout this branch.
- Create a zip for testing with `npm run test:zip`.
- Create a `jurassic.ninja` instance.
- Upload the plugin and activate it.
- Update the installation date (we need a store between 2 and 5 days old). You can do it with an SQL statement like this:

```
UPDATE `wp_options` SET `option_value`=UNIX_TIMESTAMP(DATE_SUB(NOW(), INTERVAL 4 day)) WHERE `option_name` = 'woocommerce_admin_install_timestamp';
```

- Make sure the `woocommerce_merchant_email_notifications` option is set to `yes`:
```
UPDATE `wp_options` SET `option_value` = 'yes' WHERE `wp_options`.`option_name` = 'woocommerce_merchant_email_notifications';
```

- Run the `wc_admin_daily ` cron job (this tool can help [WP Crontrol](https://wordpress.org/plugins/wp-crontrol/)).
- You should have received an email like the image above.

No changelog (landed in 2.2.0)

ilyasfoo committed:
Add capability for merchant email note personalization, improve AddFirstProduct email note contents.

ilyasfoo committed:
Update heading

ilyasfoo committed:
Update readme and testing instructions

ilyasfoo commented:
@pmcpinto Add first product email contents are as per the screenshot, are the changes enough? Please let me know!

joshuatf commented on src/Notes/AddFirstProduct.php:
```diff
@@ -57,7 +57,8 @@ public static function get_note() {
 		$content_lines = array(
-			__( 'Nice one, you’ve created a WooCommerce store! Now it’s time to add your first product.<br/><br/>', 'woocommerce-admin' ),
+			'{greetings}<br/><br/>',
+			__( 'Nice one, you’ve created a WooCommerce store! Now it’s time to add your first product and get ready to start selling.<br/><br/>', 'woocommerce-admin' ),
```
I know this was here prior to this PR, but I think the comma here is grammatically incorrect.  Either a terminal point or a semicolon may be more appropriate here.

Also, same comment about `sprintf` for `br` tags.

joshuatf commented on TESTING-INSTRUCTIONS.md:
```diff
@@ -409,6 +409,23 @@ Scenario #2
 
+### Improve AddFirstProduct email note contents #6617
+
+- Install the plugin in a fresh site site.
```
Just to be extra safe, might be worth mentioning that the store should have 0 orders and 0 products.

Also `site site` 😄

joshuatf reviewed (commented):
Nice work, @ilyasfoo!  This is testing well, just a few clean-up comments.

ilyasfoo committed:
Remove old image, fix typo and improve translation

ilyasfoo commented:
Thanks for the review @joshuatf! I've addressed most of the feedback in 624c18a. 

While I agree with the comments to use `sprintf` on the translations to separate HTML, I'd argue that changing it right now would add unnecessary work for our translators, since it has been already translated (I've checked at least for Spanish). Instead, I added a comment in the code to let the next person know if the texts are going to change in the future. Would you agree?

pmcpinto commented:
> @pmcpinto Add first product email contents are as per the screenshot, are the changes enough? Please let me know!

Yep, I think so. @elizaan36 can you just give a quick look as well? Thanks!

elizaan36 commented:
Looks good to me👍

joshuatf commented:
Sorry for the late review on this one, @ilyasfoo.

> I'd argue that changing it right now would add unnecessary work for our translators since it has already been translated (I've checked at least for Spanish).

That's probably a fair assessment.  LGTM 🎉

ilyasfoo commented:
I think you forgot to approve @joshuatf 😄

joshuatf reviewed (approved):
Weird, I thought I approved.  LGTM still! 🚢
